### ISSUE TITLE
[bug]: Entire app freezing with no input possible after closing WalletConnect Modal

### ISSUE THREAD

**Issue by musnit** [WalletConnect/react-native-examples#78]
### Description

Open the modal, then tap to close it and thereafter the app freezes. 

### Issue Environment

- [ ] dapps/v2Explorer
- [ ] wallets/rn_cli_wallet
- [ ] wallets/rn_cli_wallet_068_5
- [ ] wallets/expo_wallet
- [ ] wallets/web3wallet_tutorial
- [X] In my own code

### Output of `npx react-native info`

System:
  OS: macOS 13.4.1
  CPU: (10) x64 Apple M1 Max
  Memory: 532.38 MB / 64.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 16.17.0
    path: ~/.nvm/versions/node/v16.17.0/bin/node
  Yarn:
    version: 1.22.18
    path: ~/.yarn/bin/yarn
  npm:
    version: 9.2.0
    path: ~/.nvm/versions/node/v16.17.0/bin/npm
  Watchman:
    version: 2023.01.16.00
    path: /usr/local/bin/watchman
Managers:
  CocoaPods:
    version: 1.11.3
    path: /Users/aidanmusnitzky/.rvm/gems/ruby-3.0.0/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 22.4
      - iOS 16.4
      - macOS 13.3
      - tvOS 16.4
      - watchOS 9.4
  Android SDK: Not Found
IDEs:
  Android Studio: 2021.1 AI-211.7628.21.2111.8309675
  Xcode:
    version: 14.3.1/14E300c
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 11.0.16
    path: /Users/aidanmusnitzky/.jenv/shims/javac
  Ruby:
    version: 3.0.0
    path: /Users/aidanmusnitzky/.rvm/rubies/ruby-3.0.0/bin/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.72.0
    wanted: 0.72.0
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: true
  newArchEnabled: false

### Steps to reproduce

This is occurring on a even a really basic app with just:

```
  return (
    <>
      <WalletConnectModal
        projectId={config.WALLETCONNECT_PROJECT_ID}
        providerMetadata={providerMetadata}
      />
      <View style={{flex: 1, justifyContent: 'center', alignItems: 'center'}}>
        <Button title="Press me" onPress={connect} />
      </View>
      <AppBootstrap />
    </>
  );
};
```

### Snack, code example, screenshot, or link to a repository

https://github.com/musnit/broken_walletconnect

**Comment by musnit** [comment]
This seems related to react version. When switching from 0.71.8 as in your working demo to 0.72, it breaks

**Comment by musnit** [comment]
The library also does not work on 0.67.5 with the same issue - presumably the same cause

**Comment by ignaciosantise** [comment]
hey @musnit i've just tested with 0.72 and it breaks in iOS. I'll dig into it to see if i can solve this.

**Comment by musnit** [comment]
> hey @musnit i've just tested with 0.72 and it breaks in iOS. I'll dig into it to see if i can solve this.

thanks! it's less critical for us now since we're using our own fork with our own UI which slices out all UI code (https://github.com/spinamp/walletconnect-react-native)

**Comment by ignaciosantise** [comment]
@musnit i found the issue, it's related with some LayoutAnimations. Will release the fix tomorrow.

On the other hand, we currently don't have plans of creating a headless sdk in the mid-term but we are working to make the UI more customizable. Can you help us understand what you want to customize and what is important to you?

For example:
- Do you want a particular design? (eg list vs grid?)
- Change colors and layout?
- Can you share a screenshot/mock of how you are customizing it?

**Comment by musnit** [comment]
> @musnit i found the issue, it's related with some LayoutAnimations. Will release the fix tomorrow.
> 
> On the other hand, we currently don't have plans of creating a headless sdk in the mid-term but we are working to make the UI more customizable. Can you help us understand what you want to customize and what is important to you?
> 
> For example:
> 
> * Do you want a particular design? (eg list vs grid?)
> * Change colors and layout?
> * Can you share a screenshot/mock of how you are customizing it?

We've already created a complete working SDK and it's live - I'd suggest taking a look and maybe just sharing that with others too: https://github.com/spinamp/walletconnect-react-native

We want (and have) full customizability over everything with no provided UI whatsoever.

We're still waiting on Apple review to approve the new version of our app, but once it's live I can share a link for you to demo it.

**Comment by ignaciosantise** [comment]
Great, i'll take a look at it 💪 

Closing this issue, as it's solved in `@walletconnect/modal-react-native@1.0.0-rc.6`
Thanks for reporting @musnit
